cloud/etc/deploy-microovn/main.tf

```diff
@@ -0,0 +1,113 @@
+# microovn.tf
+terraform {
+  required_providers {
+    juju = {
+      source  = "juju/juju"
+      version = "= 0.23.0"
+    }
+  }
+}
+
+provider "juju" {}
+
+resource "juju_application" "openstack-network-agents" {
+  name  = "openstack-network-agents"
+  trust = true
+  model = var.machine_model
+
+  charm {
+    name    = "openstack-network-agents"
+    channel = var.charm_openstack_network_agents_channel
+    base    = "ubuntu@24.04"
+  }
+}
+
+resource "juju_application" "microcluster-token-distributor" {
+  name  = "microcluster-token-distributor"
+  trust = true
+  model = var.machine_model
+  machines = length(var.machine_ids) == 0 ? null : toset(var.machine_ids)
```

Comment:
Can we have separate variable for machine_ids for token-distributor, reason being in future all network+compute nodes will have microovn but we dont need token-distributor on all nodes.